src/Dds/Services/CrudService.cs

```diff
@@ -83,22 +83,38 @@ public ReadResponse Read(string storeName, int start, int pageSize, string searc
             var store = DynamicDataStoreFactory.Instance.GetStore(storeName);
             var storeMetadata = store.Metadata();
 
-            // TODO: we cannot order here due to fact that this is PropertyBag, if we could it would be great performance boost
-            // var orderBy = sortByColumn == 0 ? "Id" : StoreMetadata.Columns.ToList()[sortByColumn - 1].PropertyName;
-            var query = store.ItemsAsPropertyBag(); // .OrderBy(orderBy);
+            List<PropertyBag> data;
+            int count;
 
-            var data = sortByColumn == 0 && string.IsNullOrEmpty(search)
-                               ? (sortDirection == "asc"
-                                          ? query.OrderBy(r => r.Id).Skip(start).Take(pageSize).ToList()
-                                          : query.OrderByDescending(r => r.Id).Skip(start).Take(pageSize).ToList())
-                               : query.ToList();
+            var preSorted = false;
+
+            if (string.IsNullOrWhiteSpace(filterColumnName) || string.IsNullOrWhiteSpace(filter))
+            {
+                // TODO: we cannot order here due to fact that this is PropertyBag, if we could it would be great performance boost
+                // var orderBy = sortByColumn == 0 ? "Id" : StoreMetadata.Columns.ToList()[sortByColumn - 1].PropertyName;
+                var query = store.ItemsAsPropertyBag(); // .OrderBy(orderBy);
+
+                preSorted = sortByColumn == 0 && string.IsNullOrEmpty(search);
```

Comment:
I think that `search` / `filter` does not work as anyone would expect when used together.